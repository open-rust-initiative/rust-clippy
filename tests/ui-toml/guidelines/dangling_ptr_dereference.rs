#![allow(unused)]
#![warn(clippy::dangling_ptr_dereference)]

fn my_free(_ptr: *mut i8) {}

fn main() {
    unsafe {
        let ptr: *mut i8 = 1 as *mut _;
        my_free(ptr);
        let _ = *ptr; // lint

        let ptr_1: *mut i8 = 1 as *mut _;
        my_free(ptr_1);
        println!("{:?}", *ptr_1); // lint

        let ptr_2: *mut i8 = 1 as *mut _;
        my_free(ptr_2);
        if !ptr_2.is_null() {
            println!("{:?}", *ptr_2); // don't lint
        }
    }
}
