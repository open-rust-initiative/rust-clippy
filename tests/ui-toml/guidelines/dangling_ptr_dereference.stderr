error: dereferencing a raw pointer that was already freed
  --> $DIR/dangling_ptr_dereference.rs:10:17
   |
LL |         let _ = *ptr; // lint
   |                 ^^^^
   |
note: the pointer was freed here
  --> $DIR/dangling_ptr_dereference.rs:9:9
   |
LL |         my_free(ptr);
   |         ^^^^^^^^^^^^
   = note: `-D clippy::dangling-ptr-dereference` implied by `-D warnings`

error: dereferencing a raw pointer that was already freed
  --> $DIR/dangling_ptr_dereference.rs:14:26
   |
LL |         println!("{:?}", *ptr_1); // lint
   |                          ^^^^^^
   |
note: the pointer was freed here
  --> $DIR/dangling_ptr_dereference.rs:13:9
   |
LL |         my_free(ptr_1);
   |         ^^^^^^^^^^^^^^

error: dereferencing a raw pointer that was already freed
  --> $DIR/dangling_ptr_dereference.rs:19:30
   |
LL |             println!("{:?}", *ptr_2); // don't lint
   |                              ^^^^^^
   |
note: the pointer was freed here
  --> $DIR/dangling_ptr_dereference.rs:17:9
   |
LL |         my_free(ptr_2);
   |         ^^^^^^^^^^^^^^

error: aborting due to 3 previous errors

